src/Http/ServerRequest.php

```diff
@@ -380,12 +380,16 @@ protected function _processGet(array $query, string $queryString = ''): array
     /**
      * Process uploaded files and move things onto the post data.
      *
-     * @param array $post Post data to merge files onto.
+     * @param mixed $post Post data to merge files onto.
      * @param array $files Uploaded files to merge in.
      * @return array merged post + file data.
      */
     protected function _processFiles($post, array $files)
```

Comment:
When can this ever not operate on array? I checked, the code before for sure returns an array

    $config['post'] = $this->_processPost($config['post']);

As such I wonder if we should even allow for this case here and weaken the API?

Reply:
There's a test case where POST data is string:

https://github.com/cakephp/cakephp/blob/b1673c25d8955cc69e551fa97e7a159bb335f82a/tests/TestCase/Http/ServerRequestTest.php#L2437-L2443

I can change the `mixed` to `array|string` if you prefer.

Reply:
sounds fair.